### ISSUE TITLE
[Tune] add to the Tuner class interface a mean to get back the tune_config and run_config from a restored tuner

### ISSUE THREAD

**Issue by Edouard59** [ray-project/ray#41364]
### Description

When restoring a tuner with Tuner.restore(), there is as of today no public interface to get back the tune_config and run_config parameters that you normally provide when creating a tuner via Tuner.Tuner(). Having it available at least on the run_config can be useful to get back the instances of classes provided as callbacks, in their restored state, in case those classes are used after the tuning.

### Use case

I'm providing to my tuner a custom logger class (derived from LoggerCallback) via the run_config.callbacks parameter. In my application, that provided class does some stuff during the tuning (log_trial_start, log_trial_end...) which will update some of its internal members. After tuning is complete, I process the results from the tuning and then after that and based on this processing, I want to call another function from my logger class to process them furter, using the scope it accumulated during logging. All this works fine except when I want to go with restoring a tune run. Because in that case I cannot get a handle back onto my restored logger class. One way I found today to do it is to call tuner._local_tuner.get_run_config(). Would be nice to make it a public interface of the Tuner instead.

A bit of code do demonstrate the discussed use case and how I got it to work today through the private interface:

`
    if tune.Tuner.can_restore(exp_dir):
        tuner = tune.Tuner.restore(
            exp_dir, trainable=myTrainable, resume_errored=True
        )
        # FIXME: kind of a hack to get back a hand onto the logger class
        runConfig = tuner._local_tuner.get_run_config()
        for callback in runConfig.callbacks:
            if isinstance(callback, MyLoggerClass):
                myLogger= callback
    else:
        myLogger = MyLoggerClass(someData)
        tuner = tune.Tuner(
            trainable=myTrainable,
            param_space=params["tuner"]["param_space"],
            tune_config=tune.TuneConfig(**params["tuner"]["tune_config"]),
            run_config=train.RunConfig(callbacks=[myLogger]),
        )

    results = tuner.fit()
    Process(results)
    myLogger.DoSomeStuff()

`

**Comment by matthewdeng** [comment]
I think this makes sense, @justinvyu any concerns if we add this as public attributes?

**Comment by justinvyu** [comment]
Yeah, I think this is ok to add. However, I think that accessing the restored loggers from the run config do not keep any state from the run. Neither are the callbacks unless you call `callback.restore_from_dir` manually.
